### PR TITLE
Add Minimum Separation to Hinge Loss

### PR DESCRIPTION
**Pull request by lilleswing**
Something the author did in ScScore is change the minimum separation of the hinge-loss from 1.0 to 0.25 for ScScore.

Since the scores range from 1-5 this meant that there could be more levels of differentiation for compounds in the reaction set.  This is an important hyper parameter to tune if you are doing synthetic feasibility over a large reaction corpus.

### CODE REVIEW

File: deepchem/models/tensorgraph/layers.py [hunk 1]
```diff
@@ -4622,9 +4634,13 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
     if len(inputs) != 2:
       raise ValueError()
     labels, logits = inputs[0], inputs[1]
-    reduction = tf.losses.Reduction
-    out_tensor = tf.losses.hinge_loss(
-        labels=labels, logits=logits, reduction=reduction.NONE)
+
+    all_ones = array_ops.ones_like(labels)
+    labels = tf.cast(math_ops.subtract(2 * labels, all_ones), dtype=tf.float32)
+    seperation = tf.multiply(
```
[the remark this answers is not in the document]
Comment (lilleswing) [reply deepchem/models/tensorgraph/layers.py]: nope just can't spell
